### ISSUE TITLE
RabbitMQ in docker compose file is very out of date

### ISSUE THREAD

**Issue by Montg0mery** [zulip/docker-zulip#422]
From `docker-compose.yaml`:

```
  rabbitmq:
    image: "rabbitmq:3.7.7"
```

You can see from here (https://www.rabbitmq.com/versions.html) that RabbitMQ 3.7 went end of life in 2020.

Unless there is a specific reason to be pinned to an end-of-life version (and a specific patch-level at that), please update this to the latest stable version (and don't pin to a specific patch-level if there's no reason to).

Thank you.

**Comment by timabbott** [comment]
Yeah, there's no good reason for that to be pinned to an old version. @mikkeschiren would you have time to test the latest version and put up a PR? I would also not mind this changing to a `latest` tag; we don't depend on anything specific about the version.

**Comment by SGNayak12** [comment]
Hi Zulip Docker Maintainers,

I hope you’re doing well!

I noticed that the docker-compose.yml file in the zulip-docker repository is currently using RabbitMQ version 3.12.14, which is now unsupported(https://www.rabbitmq.com/release-information). The latest stable version is 4.0.6, and I was wondering if you would be open to upgrading the RabbitMQ version in the repository to this latest version.

Please let me know if this sounds good, or if you are already planning to update it.

**Comment by galexrt** [comment]
> [...] I would also not mind this changing to a `latest` tag [...]

It is recommended use a (more) specific tag/version even if that means that it needs to be updated from time to time, in case something breaks it is easier to debug when it is clear what version of an image is/was used.
A compromise of both worlds, that I have personally started using for (public) projects, is to use "partial specific version tags", e.g., to get RabbitMQ version `4.0.x` one could use the tag `4.0` (see screenshot).

![Image](https://github.com/user-attachments/assets/9d875630-3e4f-49c8-938c-3370a843ef25)

***

@SGNayak12 I'm not sure why you sent me an email regarding this issue/your comment on this issue, I haven't worked on this project/repository since like 2018...

**Comment by SGNayak12** [comment]
In CODE_OF_CONDUCT.md this email id was given in case of any queries
that is why i contacted you

**Comment by SGNayak12** [comment]
By the way it would be great if you give some tips for open source
contribution for upcoming GSCO'25.

**Comment by alexmv** [comment]
3677a238d3c67f5a8e929e8f3c9a8a81e2087263 moved this to be 4.0.7, and I just pushed 5ffcac8023884ce8e887d42089af3221828bde62 to make it the latest 4.1.x.
